microsetta_interface/implementation.py

```diff
@@ -813,25 +812,89 @@ def get_create_human_source(*, account_id=None):
     if has_error:
         return consent_output
 
+    form_type = "source"
+
     return _render_with_defaults(
         'new_participant.jinja2',
         tl=consent_output,
         post_url=post_url,
         duplicate_email_check_url=duplicate_email_check_url,
         home_url=home_url,
+        form_type=form_type,
         language_tag=session_locale())
 
 
 @prerequisite([ACCT_PREREQS_MET])
 def post_create_human_source(*, account_id=None, body=None):
-    has_error, consent_output, _ = ApiRequest.post(
-        "/accounts/{0}/consent".format(account_id), json=body)
-    if has_error:
-        return consent_output
 
-    new_source_id = consent_output["source_id"]
+    create_new_source = False
+    consent_type = body.get("consent_type")
+
+    if "Data" in consent_type:
+        consent_type = "Data"
+
+        if "source_id" in session:
+            source_id = session['source_id']
+
+            dup_check_body = {}
+            dup_check_body["participant_name"] = body.get("participant_name")
+
+            has_error, email_check_output, _ = ApiRequest.post(
+                "/accounts/{0}/check_duplicate_source".format(
+                    account_id), json=dup_check_body)
```

Comment:
This code should be removed. We're only checking the duplicate source name on the interface as a suggestion to not create duplicates, not a requirement. And if the source_id is in session - meaning they've already created a source, then this is always going to find an existing source with the same name.

Reply:
Corrected!